### PR TITLE
IGNITE-18523 Destroy RocksDbTableStorage on destroy()

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-18523

### CODE REVIEW

File: modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/AbstractMvTableStorageTest.java [hunk 1]
```diff
@@ -544,37 +544,30 @@ public void testStartRebalanceForClosedPartition() {
...
-        IndexRow hashIndexRow = indexRow(hashIndexStorage.indexDescriptor(), tableRow, rowId);
-        IndexRow sortedIndexRow = indexRow(sortedIndexStorage.indexDescriptor(), tableRow, rowId);
-
-        mvPartitionStorage.runConsistently(() -> {
-            mvPartitionStorage.addWriteCommitted(rowId, tableRow, clock.now());
-
-            hashIndexStorage.put(hashIndexRow);
-
-            sortedIndexStorage.put(sortedIndexRow);
+        List<IgniteTuple3<RowId, TableRow, HybridTimestamp>> rows = List.of(
```
Comment (ibessonov) [inline modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/AbstractMvTableStorageTest.java:552]: Why does it have to be a list of tuples? Looks ugly.
Why a single entry wasn't enough?

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/HashIndex.java [hunk 2]
```diff
@@ -83,4 +83,12 @@ void destroy(int partitionId, WriteBatch writeBatch) throws RocksDBException {
     @Nullable RocksDbHashIndexStorage get(int partitionId) {
         return storages.get(partitionId);
     }
+
+
```
Comment (ibessonov) [inline modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/HashIndex.java:87]:
```suggestion

```

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/HashIndex.java [hunk 3]
```diff
@@ -83,4 +83,12 @@ void destroy(int partitionId, WriteBatch writeBatch) throws RocksDBException {
...
+    /**
+     * Closes all index storages.
+     */
+    void close() {
+        storages.values().forEach(RocksDbHashIndexStorage::close);
```
Comment (ibessonov) [inline modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/HashIndex.java:92]: Why don't you use "closeAll"?

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/SortedIndex.java [hunk 4]
```diff
@@ -90,6 +90,8 @@ void destroy(int partitionId, WriteBatch writeBatch) throws RocksDBException {
 
     @Override
     public void close() {
+        storages.values().forEach(RocksDbSortedIndexStorage::close);
+
         indexCf.handle().close();
```
Comment (ibessonov) [inline modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/SortedIndex.java:95]: Why don't you use "closeAll" here as well?

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/HashIndex.java (hunk 2 above)
Comment (tkalkirill) [reply modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/HashIndex.java:87]: Fix it

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/SortedIndex.java (hunk 4 above)
Comment (tkalkirill) [reply modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/SortedIndex.java:95]: Fix it.

File: modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/AbstractMvTableStorageTest.java (hunk 1 above)
Comment (tkalkirill) [reply modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/AbstractMvTableStorageTest.java:552]: These are tests, they(`IgniteTuple3`) are already used in several tests, let's not touch them?

**Review by ibessonov — APPROVED** [review]

**Pull request merged**